content/doc/administration/requirements/linux.adoc

```diff
@@ -0,0 +1,78 @@
+---
+layout: documentation
+title:  Linux Support Policy
+---
+
+This page documents the Linux support policy for the Jenkins controller and agents.
+
+== Scope
+
+Jenkins plugins may set additional requirements to Linux versions on controllers and/or agents.
+This page does not document such requirements.
+Please refer to plugin documentation.
+
+== Why?
+
+Theoretically, Jenkins can run everywhere where you can run a supported Java version,
+but there are some limitations in practice.
+Jenkins core and some plugins include native code or depend on Linux API and subsystems,
+and hence they rely on specific Linux versions.
+Jenkins platform specific installation packages rely on specific Linux versions.
+
+== Support levels
+
+We define multiple support levels for Linux platforms.
+
+[width="100%",cols="20%,35%,45%",options="header",]
+|===
+|Support level |Description |Platforms
+
+| **Level 1** - Full support
+| We run automated testing for these platforms, and we intend to fix the reported issues timely.
+a|
+  * 64-bit (amd64) Linux versions supported by Debian, Red Hat, SUSE, and Ubuntu
+  * Linux versions used in the official Docker images
+
+| **Level 2** - Supported
+| We do not actively test these platforms, but we intend to keep compatibility.
```

Comment:
> We do not actively test these platforms, but we intend to keep compatibility.

How do we intend to keep compatibility when making large-scale changes if we do not actively test these platforms?

Reply:
> How do we intend to keep compatibility when making large-scale changes if we do not actively test these platforms?

We rely on users and contributors to test these platforms.  When they encounter an issue, they report it.

In the case of some of the platforms in the list (like arm64 and s390x on Debian), we are actively testing them with the [weekly installation acceptance test job](https://ci.jenkins.io/job/Infra/job/acceptance-tests/job/install-latest-debian-package/) and the [LTS installation acceptance test job](https://ci.jenkins.io/job/Infra/job/acceptance-tests/job/install-lts-debian-package/).  However, that little bit of installation testing was not enough for me to feel that we should list those platforms as "Level 1".

Reply:
> We rely on users contributors to test these platforms

So a contributor making a large-scale change would be expected to test that large-scale change on Level 2 platforms (or solicit testing by users and debug by telegram) prior to integration, correct? That does not seem congruent with the notion of "we do not actively test these platforms". And who is the "we" referring to, anyway? Maintainers? Are maintainers not contributors, too?

Reply:
> > We rely on users and contributors to test these platforms
> 
> So a contributor making a large-scale change would be expected to test that large-scale change on Level 2 platforms (or solicit testing by users and debug by telegram) prior to integration, correct? That does not seem congruent with the notion of "we do not actively test these platforms". And who is the "we" referring to, anyway? Maintainers? Are maintainers not contributors, too?

I should have been more precise.  I should have said.

> We rely on users to test these platforms

I was using an overly broad definition of "contributor" that is indistinguishable from "user".  Should have kept the phrasing simple and said "users" instead of "users and contributors".

Reply:
Got it. Then a contributor (who may or may not also be a maintainer) making a large-scale change would have to rely on users to test the change on Level 2 platforms (and possibly debug issues by telegram) before the change could be integrated. This seems undesirable to me.

Reply:
> Then a contributor (who may or may not also be a maintainer) making a large-scale change would have to rely on users to test the change on Level 2 platforms (and possibly debug issues by telegram) before the change could be integrated. This seems undesirable to me.

The policy does not limit what a contributor can do.  It tries to describe the minimum levels we are striving to do.

A contributor is welcome to do more testing before the change is integrated.

Reply:
Of course, but if a contributor is _required_ to either do that testing or to solicit users to do it on the contributor's behalf, then IMHO the platform is "fully" supported.